Feature/authentication module

Zlatonn opened the pull request:
### **Features Included**
- login feature
- registor feature
- implement auth guard

_registor_
<img width="1030" height="622" alt="image" src="https://github.com/user-attachments/assets/1d2f4f7e-a2bd-403b-b9e5-bacb750d269f" />

_login_
<img width="1039" height="579" alt="image" src="https://github.com/user-attachments/assets/5336d260-064a-400d-ada8-8a9fa00b3b3a" />

Zlatonn committed:
apply hash password for seeding

Zlatonn committed:
implement feature login

- create auth module
- create login dto
- create login controller
- create login service

Zlatonn committed:
implement feature registor

- create registor dto
- create registor controller
- create registor service

Zlatonn committed:
implement auth guard

- create auth guard
- crate skip auth decorators

looksocii commented on src/modules/auth/auth.service.ts:
```diff
@@ -0,0 +1,74 @@
+    };
+  }
+
+  async registor(req: RegistorDTO) {
```
register

looksocii commented on src/modules/auth/auth.service.ts:
```diff
@@ -0,0 +1,74 @@
+      role,
+    };
+
+    return await this.prismaService.user.create({ data: userData });
```
omit password ออกด้วยนะครับ

Zlatonn committed:
update isHash DTO

- update DTO that support hash sha1
- update seeding

Zlatonn committed:
request change#2

- add omit for register response
- change word registor => register
- change word RegisterDTO => RegisterDto
